cmake/KokkosConfig.cmake.in

```diff
@@ -19,7 +19,17 @@ include("${Kokkos_CMAKE_DIR}/KokkosTargets.cmake")
 include("${Kokkos_CMAKE_DIR}/KokkosConfigCommon.cmake")
 unset(Kokkos_CMAKE_DIR)
 
-# check for conflicts
+# components deprecation
+if("launch_compiler" IN_LIST Kokkos_FIND_COMPONENTS)
+  message(WARNING "'launch_compiler' is deprecated and will be removed in the future. Use the 'Kokkos_FORCE_COMPILER_LAUNCHER' option instead.")
+  set(Kokkos_GLOBAL_COMPILATION true)
+  set(Kokkos_FORCE_COMPILER_LAUNCHER true)
+elseif("separable_compilation" IN_LIST Kokkos_FIND_COMPONENTS)
+  message(WARNING "'separable_compilation' is deprecated and will be removed in the future. Use the 'Kokkos_GLOBAL_COMPILATION' option instead.")
```

Comment:
```suggestion
  message(DEPRECATION "'separable_compilation' is deprecated and will be removed in the future. Use the 'Kokkos_GLOBAL_COMPILATION' option instead.")
```

Reply:
I think we rather want a `WARNING` here. The `DEPRECATION` stuff is an opt-in print